lib/utmp.c

```diff
@@ -368,12 +367,13 @@ setutmp(struct utmpx *ut)
 
 
 int
-update_utmp(const char *user, const char *tty, const char *host)
+update_utmp(const char *user, const char *tty, const char *host,
+            pid_t main_pid)
 {
 	struct utmpx  *utent, *ut;
 
-	utent = get_current_utmp ();
-	ut = prepare_utmp  (user, tty, host, utent);
+	utent = get_current_utmp (main_pid);
+	ut = prepare_utmp  (user, tty, host, utent, main_pid);
```

Comment:
Please remove the space before the `(`.  I take the opportunity of every change in existing function calls to remove those (since that doesn't complicate the diff in any way).

Reply:
OK